### PR TITLE
Adding GitHub PAT to url to enable pulling from private repos

### PR DESCRIPTION
**Pull request by fbbotero-aws**
*Description of changes:*
Adding support to pull from private GitHub repositories by using PAT authentication and HTTPS urls

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: ci/snapshot/prepare_source.py [hunk 1]
```diff
@@ -446,6 +446,16 @@ def generate_cbmc_jobs(src, repo_id, repo_sha, is_draft, tarfile, logger):
         raise pending_exception
 
 ################################################################
+SECRET_TARGET_GITHUB_PAT_NAME = 'GitHubCommitStatusPAT'
+def format_github_url(url, secret_manager = boto3.client('secretsmanager')):
+    secret = secret_manager.get_secret_value(SecretId=SECRET_TARGET_GITHUB_PAT_NAME)
+    if secret is not None:
+        parsed_url = urlparse(url)
+        if parsed_url.scheme == 'https':
+            token = str(json.loads(secret['SecretString'])[0]['GitHubPAT'])
```
Comment (eidelmanjonathan) [inline ci/snapshot/prepare_source.py:455]: Nitpick: maybe should be string constants

File: ci/snapshot/prepare_source.py [hunk 2]
```diff
@@ -457,7 +467,8 @@ def source_prepare():
         logging.debug(debug_json('invocation', script_data(arg)))
         cbmc_ci_github.update_status("pending", "Proof jobs starting", None, "Status pending", arg.id, arg.sha, False)
         base_name = repository_basename(arg.repository)
-        clone_repository(arg.repository, base_name)
+        repository_url = format_github_url(arg.repository)
+        clone_repository(repository_url, base_name)
```
Comment (eidelmanjonathan) [inline ci/snapshot/prepare_source.py:471]: Please put a fixme that we want to hide the token better in the logs

Comment (fbbotero-aws) [reply ci/snapshot/prepare_source.py:473]: Done

File: ci/snapshot/prepare_source.py (hunk 1 above)
Comment (fbbotero-aws) [reply ci/snapshot/prepare_source.py:456]: Added FIXME

**Pull request merged**